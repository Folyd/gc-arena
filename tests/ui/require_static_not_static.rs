use gc_arena::Collect;

struct NoCollectImpl<'a>(&'a bool);

#[derive(Collect)]
#[collect(no_drop)]
struct MyStruct<'a> {
    #[collect(require_static)]
    field: NoCollectImpl<'a>
}

fn assert_my_struct_collect<'a>() {
    MyStruct::<'a>::needs_trace();
}

fn main() {}
