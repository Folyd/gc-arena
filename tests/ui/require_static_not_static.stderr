error: lifetime may not live long enough
  --> $DIR/require_static_not_static.rs:13:5
   |
12 | fn assert_my_struct_collect<'a>() {
   |                             -- lifetime `'a` defined here
13 |     MyStruct::<'a>::needs_trace();
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^ requires that `'a` must outlive `'static`
